DigitalOffice/Tests/CompanyServiceTests/Repositories/PositionRepositoryTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using LT.DigitalOffice.CompanyService.Database;
+using LT.DigitalOffice.CompanyService.Database.Entities;
+using LT.DigitalOffice.CompanyService.Repositories;
+using LT.DigitalOffice.CompanyService.Repositories.Interfaces;
+using LT.DigitalOffice.CompanyServiceUnitTests.UnitTestLibrary;
+using Microsoft.EntityFrameworkCore;
+using NUnit.Framework;
+
+namespace LT.DigitalOffice.CompanyServiceUnitTests.Repositories
+{
+    internal class PositionRepositoryTests
+    {
+        private const string PositionName = "Software Engineer";
+        private const string Description = "some description";
+        private DbCompanyUser dbCompanyUser;
+        private CompanyServiceDbContext dbContext;
+        private DbPosition dbPosition;
+        private IPositionRepository repository;
+
+        private CompanyServiceDbContext GetMemoryContext()
+        {
+            var options = new DbContextOptionsBuilder<CompanyServiceDbContext>()
+                .UseInMemoryDatabase("InMemoryDatabase")
+                .Options;
+
+            return new CompanyServiceDbContext(options);
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            dbContext = GetMemoryContext();
+            repository = new PositionRepository(dbContext);
+
+            dbPosition = new DbPosition
+            {
+                Id = Guid.NewGuid(),
+                Description = Description,
+                Name = PositionName,
+                UserIds = new List<DbCompanyUser>()
+            };
+
+            dbCompanyUser = new DbCompanyUser
+            {
+                PositionId = dbPosition.Id,
+                UserId = Guid.NewGuid()
+            };
+
+            dbPosition.UserIds.Add(dbCompanyUser);
+
+            dbContext.Positions.Add(dbPosition);
+            dbContext.SaveChanges();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            if (dbContext.Database.IsInMemory()) dbContext.Database.EnsureDeleted();
+        }
+
+        #region GetUserPosition definition
+
+        [Test]
+        public void ShouldThrowExceptionWhenUserIdEmpty()
+        {
+            Assert.Throws<Exception>(() => repository.GetUserPosition(Guid.Empty));
+        }
+
+        [Test]
+        public void ShouldReturnUserPosition()
+        {
+            var expected = new DbPosition
+            {
+                Name = PositionName,
+                Description = Description,
+                Id = dbPosition.Id
+            };
+
+            SerializerAssert.AreEqual(expected.Id, repository.GetUserPosition(dbCompanyUser.UserId).Id);
+            SerializerAssert.AreEqual(expected.Description, repository.GetUserPosition(dbCompanyUser.UserId).Description);
+            SerializerAssert.AreEqual(expected.Name, repository.GetUserPosition(dbCompanyUser.UserId).Name);
```

Comment:
Also, add checks for db state. Something like that: `Assert.That(dbContext.Positions, Is.EquivalentTo(new[] { expected})) `

Reply:
1. I can't use SerializerAssert, because JsonSerialiser can't serialize because it throws an error about the potential loop.
2. I don't understand why. I do not change db state.

Reply:
1. But you are using SerializerAssert. Why are not you using simple Assert?
2. Well, why not make sure of it. Incorrect database manipulation can cause enormous damage.

Reply:
1. My bad. fixed
2. Added assert for that